File: 3859-maximum-product-of-two-digits/maximum-product-of-two-digits.py
class Solution:
    def maxProduct(self, n: int) -> int:
        num_str = str(n)
        largest = -1
        second_largest = -1
        for char in num_str:
            digit = int(char)
            if digit > largest:
                second_largest = largest
                largest = digit
            elif digit > second_largest:
                second_largest = digit
        return largest * second_largest

File: 3859-maximum-product-of-two-digits/maximum-product-of-two-digits.cpp
class Solution {
public:
    int maxProduct(int n) {
        string s = to_string(n);
        int mx = -1;
        int smx = -1;
        for (char c : s) {
            int d = c - '0';
            if (d > mx) {
                smx = mx;
                mx = d;
            } else if (d > smx) {
                smx = d;
            }
        }
        return mx * smx;
    }
};